account_operating_unit/models/account_move.py

```diff
@@ -0,0 +1,150 @@
+
+    @api.model
+    def create(self, vals):
+        if vals.get('move_id', False):
+            move = self.env['account.move'].browse(vals['move_id'])
+            if move.operating_unit_id:
+                vals['operating_unit_id'] = move.operating_unit_id.id
+        return super(AccountMoveLine, self).create(vals)
+
+    @api.model
+    def _query_get(self, domain=None):
+        if domain == None:
+            domain = []
+        if self._context.get('operating_unit_ids', False):
+            domain.append(('operating_unit_id', 'in',
+                           self._context.get('operating_unit_ids')))
+        return super(AccountMoveLine, self)._query_get(domain)
+
+    @api.multi
+    @api.constrains('operating_unit_id', 'company_id')
+    def _check_company_operating_unit(self):
+        for rec in self:
+            if rec.company_id and rec.operating_unit_id and \
+                            rec.company_id != rec.operating_unit_id.company_id:
+                raise UserError(_('Configuration error!\nThe Company in the'
+                                  ' Move Line and in the Operating Unit must '
+                                  'be the same.'))
+
+    @api.multi
+    @api.constrains('operating_unit_id', 'move_id')
+    def _check_move_operating_unit(self):
+        for rec in self:
+            if (rec.move_id and rec.move_id.operating_unit_id and
+                rec.operating_unit_id and
+                rec.move_id.operating_unit_id != self.operating_unit_id
+            ):
+                raise UserError(_('Configuration error!\nThe Operating Unit in'
+                                  ' the Move Line and in the Move must be the'
+                                  ' same.'))
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    operating_unit_id = fields.Many2one('operating.unit',
+                                        'Default operating unit',
+                                        help="This operating unit will "
+                                             "be defaulted in the move lines.")
+
+    @api.multi
+    def _prepare_inter_ou_balancing_move_line(self, move, ou_id,
+                                              ou_balances):
+        if not move.company_id.inter_ou_clearing_account_id:
+            raise UserError(_('Error!\nYou need to define an inter-operating\
+                unit clearing account in the company settings'))
+
+        res = {
+            'name': 'OU-Balancing',
+            'move_id': move.id,
+            'journal_id': move.journal_id.id,
+            'date': move.date,
+            'operating_unit_id': ou_id,
+            'account_id': move.company_id.inter_ou_clearing_account_id.id
+        }
+
+        if ou_balances[ou_id] < 0.0:
+            res['debit'] = abs(ou_balances[ou_id])
+
+        else:
+            res['credit'] = ou_balances[ou_id]
+        return res
+
+    @api.multi
+    def _check_ou_balance(self, move):
```

Comment:
When the OU is not balanced because there is more credit than debit, the balance is negative, so if when we want to balance we create a debit entry with the result of that balance. As the number is negative the absolute value is taken.